scripts/core/editor3/components/MultipleHighlights.jsx

```diff
@@ -119,70 +119,132 @@ export class MultipleHighlights extends React.Component {
         return state;
     }
 
-    addHighlight(highlightType, highlightData) {
-        // NICE-TO-HAVE: remove associated data when highlights are removed by deleting characters in the editor
+    setHighlightsState(editorState, highlightsState) {
+        // TODO: save to first block
+        this.setState(highlightsState);
+    }
+
+    getHighlightsStates(editorState) {
+        // TODO: get from first block
+        return this.state;
+    }
+
+    saveHighlights({newEditorState, newHighlightsState}, applyChanges) {
+        this.setHighlightsState(newEditorState, newHighlightsState);
+
+        if (applyChanges) {
+            this.props.onHighlightChange(newEditorState, this.exportState());
+        }
+
+        return newEditorState;
+    }
+
+    initHighlights(editorState) {
+        const applyChanges = editorState === undefined;
```

Comment:
`editorState` shouldn't be able to be undefined at this point. If it is, you should have thrown an `Error` in the calling function. ~~I don't think the bool `applyChanges` has to exist in the first place. If this is to be stateless why wouldn't we want to always apply changes?~~ Ok, it seems `applyChanges` means a different thing than I thought. I though it's a flag for deciding if we want to push a new highlights state into the `editorState`. It seems like it's a flag if we want to call `this.props.onChange` which in turn would trigger a reducer. We want to have this configurable, but it shouldn't be checked by comparing `editorState` to `undefined`. Instead it should be passed as a parameter.

Reply:
The MultipleHighlights functions has editorState as last parameter. for now this parameter is not sent from comments re: code and it this case it will be taken from props. If we change this and add the editorState parameter we can also add the check.